brainscore_language/model_helpers/huggingface.py

```diff
@@ -75,10 +80,7 @@ def digest_text(self, text: Union[str, List[str]]) -> Dict[str, DataAssembly]:
         for part_number, text_part in enumerate(text):
             # tokenize
             context = ' '.join(text[:part_number + 1])  # build up context over items in the text
-            context_tokens = self.tokenizer(context, return_tensors="pt")
-            # keep track of tokens in current `text_part`
-            self.current_tokens = {key: value[..., number_of_tokens:] for key, value in context_tokens.items()}
-            number_of_tokens = context_tokens['input_ids'].shape[-1]
+            context_tokens, number_of_tokens = self._tokenize(context, number_of_tokens)
```

Comment:
Here, "part_number" would be each 'chunk' that we are basically interested in obtaining some output (probability, embedding, ...) from, right? so we iterate through it and build up context as we go. In case we have special tokens (such as BERTs CLS or EOS tokens), the join procedure in line 77 might be weird? I would need to sit down and debug through this to also understand how this deal with bidirectional models.

Reply:
The text is just a string input, or a list of strings, for instance `text = ['the', 'quick', 'brown fox', ...]`, and then `text_part` would be e.g. `the` or `fox` etc. So the input should not have CLS/EOS tokens, those would only be added once we get to tokenization